data/base.go

```diff
@@ -0,0 +1,256 @@
+		Internal: Internal{
+			GroupID:       datum[GroupIDField].(string),
+			ActiveFlag:    true,
+			SchemaVersion: 1, //TODO: configured ??
+			CreatedTime:   time.Now().Format(time.RFC3339),
+		},
+	}
+
+	validator.SetErrorReasons(validationFailureReasons).Struct(base, errs)
+
+	return base
+}
+
+var validationFailureReasons = validate.ErrorReasons{
+	timeStringTag:     timeStrValidationMsg,
+	timeZoneOffsetTag: fmt.Sprintf("TimezoneOffset needs to be in minutes and greater than %d and less than %d", tzValidationLowerLimit, tzValidationUpperLimit),
+}
+
+func PayloadValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	//a place holder for more through validation
+	if _, ok := field.Interface().(interface{}); ok {
+		return true
+	}
+	return false
+}
+
+func AnnotationsValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	//a place holder for more through validation
+	if _, ok := field.Interface().([]interface{}); ok {
+		return true
+	}
+	return false
+}
+
+func TimezoneOffsetValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if offset, ok := field.Interface().(int); ok {
+		if offset >= tzValidationLowerLimit && offset <= tzValidationUpperLimit {
+			return true
+		}
+	}
+	return false
+}
+
+func TimeObjectValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if timeObject, ok := field.Interface().(time.Time); ok {
+		return isTimeObjectValid(timeObject)
+	}
+	return false
+}
+
+func isTimeObjectValid(timeObject time.Time) bool {
+	return !timeObject.IsZero() && timeObject.Before(time.Now())
+}
+
+func TimeStringValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if timeString, ok := field.Interface().(string); ok {
+		return isTimeStringValid(timeString)
+	}
+	return false
+}
+
+func isTimeStringValid(timeString string) bool {
+	var timeObject time.Time
+	timeObject, err := time.Parse(time.RFC3339, timeString)
+	if err != nil {
+		timeObject, err = time.Parse("2006-01-02T15:04:05", timeString)
+		if err != nil {
+			return false
+		}
+	}
+
+	return isTimeObjectValid(timeObject)
+}
+
+func ToString(fieldName string, data interface{}, errs validate.ErrorProcessing) *string {
+	if data == nil {
+		return nil
+	}
+	aString, ok := data.(string)
+	if !ok {
+		errs.Append(validate.NewPointerError(errs.BasePath+"/"+fieldName, "Invalid type", "should be of type string"))
```

Comment:
I'm seeing a lot of `errs.Append(validate.NewPointerError(errs.BasePath+"/"+`

Would it make sense to DRY that up?